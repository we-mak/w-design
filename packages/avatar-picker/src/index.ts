export { default as AvatarPicker } from "./AvatarPicker";