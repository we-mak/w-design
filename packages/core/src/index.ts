export { default as AnchorScroll } from "./AnchorScroll";
export { default as Avatar } from "./Avatar";
export { default as AvatarGroup } from "./AvatarGroup";
export { default as Badge } from "./Badge";
export { default as Box } from "./Box";
export { default as Breadcrumbs, BreadcrumbsItem } from "./Breadcrumbs";
export { default as Button, ButtonGroup } from "./Button";
export {
  default as Card,
  CardMedia,
  CardHeader,
  CardContent,
  CardFooter,
} from "./Card";

export * from "./Chip";
export { default as Chip } from "./Chip";

export { default as Collapse } from "./Collapse";
export { default as Comment } from "./Comment";
export { default as Container } from "./Container";
export { default as CustomTag } from "./CustomTag";
export { default as DisplayElement } from "./DisplayElement";
export { default as Divider } from "./Divider";
export { default as Drawer } from "./Drawer";
export { default as Dropdown } from "./Dropdown";
export { default as EmptyView } from "./EmptyView";
export { default as Flexbox } from "./Flexbox";
export { default as Icon } from "./Icon";
export { default as Image, ImageRatio } from "./Image";
export { default as List, ListItem, ListFooter } from "./List";
export { default as Menu, MenuItem, MenuHeading, SubMenu } from "./Menu";
export { default as Modal } from "./Modal";
export { default as Navbar } from "./Navbar";

export * from "./Pagination";
export { default as Pagination } from "./Pagination";

export { default as Picture } from "./Picture";
export { default as Portal } from "./Portal";

export * from "./Popover";
export { default as Popover } from "./Popover";

export { default as ProgressLoader } from "./ProgressLoader";
export { default as Provider } from "./Provider";
export { default as SectionMessage } from "./SectionMessage";

export * from "./Spinner";
export { default as Spinner } from "./Spinner";

export * from "./Steps";
export { default as Steps } from "./Steps";

export { default as Tabs, Tab } from "./Tabs";
export { default as Typo } from "./Typo";

export * from "./PushMessage";
export { default as PushMessage } from "./PushMessage";
