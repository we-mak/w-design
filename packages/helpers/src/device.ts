// TOUCH
// react:
// "onTouchStart",
// "onTouchMove",
// "onTouchEnd"
// native:
// "touchstart",
// "touchmove",
// "touchend"
// MOUSE
// react:
//  "onMouseDown",
//  "onDragOver",
//  "onMouseMove",
//  "onMouseUp"
// native:
//  "mousedown",
//  "dragStart",
//  "mousemove",
//  "mouseup"

export const isTouchDevice = !!(
  typeof window !== "undefined" &&
  typeof navigator !== "undefined" &&
  ("ontouchstart" in window || navigator.msMaxTouchPoints > 0)
);
