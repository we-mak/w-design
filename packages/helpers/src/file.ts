/**
 * checkDataURL
 * Check if data input is base 64 data URL
 */

export function checkDataURL(str: string) {
  if (str === null) return false;
  const regex = /^\s*data:([a-z]+\/[a-z]+(;[a-z-]+=[a-z-]+)?)?(;base64)?,[a-z0-9!$&',()*+;=\-._~:@/?%\s]*\s*$/i;
  return !!str.match(regex);
}
