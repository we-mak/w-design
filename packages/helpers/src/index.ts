export {
  spin,
  dashRotate,
  slideDown,
  fadeInOut,
  mediaQ,
  hex2Rgba,
  getElementSize,
  getFontSize
} from "./styled-helpers";
export {
  randomProperty,
  getValFromObjKey,
  filterObject
} from "./object-helpers";
export { firstLetter, removeAccentMarks, setUid } from "./string-helpers";
