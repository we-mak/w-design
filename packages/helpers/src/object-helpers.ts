/**
 *****************
 * Object helpers
 *****************
 */

/**
 * randomProperty
 * Pick random property from object
 */
export const randomProperty = (obj: object) => {
  const keys = Object.keys(obj);
  return obj[keys[(keys.length * Math.random()) << 0]];
};

/**
 * getValFromObjKey
 * Get value from object key
 */
export const getValFromObjKey = (key: string, targetObject: object) =>
  targetObject[key];

/**
 * filterObject
 * filter out the object by key
 */
export const filterObject = (origin: object, removed: string[]) =>
  Object.keys(origin)
    .filter(key => !removed.includes(key))
    .reduce((obj, key) => {
      obj[key] = origin[key];
      return obj;
    }, {});

// /**
//  * mergeTwoArraysToObject
//  * Combine keys array and values array to an object
//  * @example { 'default': '#4a4a4a', 'primary': '#fff' }
//  */
// export const mergeTwoArraysToObject = (keys: Array<string>, values: Array<string>): object => {
//   let result = {};
//   for (let i in keys) {
//     result[keys[i]] = values[i];
//   }
//   return result;
// };
