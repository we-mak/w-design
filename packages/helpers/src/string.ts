/**
 *****************
 * String helpers
 *****************
 */

/**
 * @function firstLetter
 * Return first letter from a string
 */
export const firstLetter = (str: string) => {
  const matches: RegExpMatchArray | null = str.match(/\b(\w)/g);
  return matches && matches.join("");
};

/**
 * @function removeAccentMarks
 * Remove accent marks for Latin characters
 * @param str target to string to remove accents
 */
export const removeAccentMarks = (str: string) => {
  return str
    .normalize("NFD")
    .replace(/[\u0300-\u036f]/g, "")
    .replace(/đ/g, "d")
    .replace(/Đ/g, "D");
};

/**
 * setUid
 * Set a local uid base on datetime
 */
export const setUid = (name: string) => {
  const now = +new Date();
  let index = 0;
  return `${name}-${now}-${++index}`;
};
