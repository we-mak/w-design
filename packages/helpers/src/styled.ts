import { keyframes, css, CSSObject } from "styled-components";

/**
 * Keyframes
 */
export const spin = keyframes`
  100% {
    transform: rotate(360deg);
    }
`;

export const dashRotate = keyframes`
  0% {
    stroke-dasharray: 1, 200;
    stroke-dashoffset: 0;
  }
  50% {
    stroke-dasharray: 89, 200;
    stroke-dashoffset: -35px;
  }
  100% {
    stroke-dasharray: 89, 200;
    stroke-dashoffset: -124px;
  }
`;

export const slideDown = keyframes`
  0% {
    opacity: 0;
    transform: translateY(-1.6rem);
  }
  100% {
    opacity: 1;
    transform: translateY(0);
  }
`;

export const fadeInOut = keyframes`
  0% {
    opacity: .1;
  }

  50% {
    opacity: .2;
  }

  100% {
    opacity: .1;
  }
`;

/**
 * Media query
 * Usage: Add mediaQ in styled component.

const FakeInput = styled.div`
  position: absolute;
  left: 8em;
  top: 4.5em;
  ${mediaQ.xl`width: 50%`};
`
*/
const screens = {
  xxl: 90,
  xl: 80,
  lg: 64,
  md: 48,
  sm: 32,
  xs: 24
};

export const mediaQ = Object.keys(screens).reduce((acc: any, key: string) => {
  acc[key] = (params: TemplateStringsArray | CSSObject) => css`
    @media (max-width: ${screens[key]}em) {
      ${css(params)}
    }
  `;
  return acc;
}, {});

/**
 * @function hex2Rgba
 * Convert hex color to rgba color
 * credit https://stackoverflow.com/questions/5623838/rgb-to-hex-and-hex-to-rgb/#answer-11508164
 */
export const hex2Rgba = (hex: string, alpha?: number): string => {
  // Check bad hex
  // if (/^#[0-9A-F]{6}$/i.test(hex)) {
  //   throw new Error("not a valid Hex");
  // }

  hex = hex.replace("#", "");
  const bigint = parseInt(hex, 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  if (alpha) {
    return `rgba(${r}, ${g}, ${b}, ${alpha})`;
  } else {
    return `rgba(${r}, ${g}, ${b})`;
  }
};

/**
 * @function getElementSize
 * @param styleProperty
 * @param size
 */
export const getElementSize = (styleProperty: object, size?: "sm" | "md" | "lg") =>
  styleProperty[size || "md"];

/**
 * @function getFontSize
 * Call function to generate the font size from default values or theme values
 */
export const getFontSize = (elementSizeProps: "sm" | "md" | "lg", fontSizes: Array<string>) => {
  switch (elementSizeProps) {
    case "sm":
      return fontSizes[0];
    case "md":
      return fontSizes[1];
    case "lg":
      return fontSizes[2];
  }
};
