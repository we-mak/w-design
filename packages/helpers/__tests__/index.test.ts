import {
  randomProperty,
  getValFromObjKey,
  filterObject,
  firstLetter,
  removeAccentMarks
} from "../lib";
import { setUid } from "../src";

const testObject = {
  foo: "lorem",
  bar: "ipsum",
  baz: "dolor"
};

describe("object helper tests", () => {
  test("randomProperty should get the radom value from object", () => {
    expect(randomProperty(testObject)).toMatchSnapshot();
  });

  test("getValFromObjKey should get the correct value from key", () => {
    expect(getValFromObjKey("bar", testObject)).toBe("ipsum");
  });

  test("filterObject should filter out the object by keys", () => {
    expect(filterObject(testObject, ["bar"])).toMatchObject({
      foo: "lorem",
      baz: "dolor"
    });
  });
});

describe("string helper tests", () => {
  test("firstLetter should get the first letter from string", () => {
    expect(firstLetter("Lorem Ipsum")).toBe("LI");
  });

  test("removeAccentMarks remove accent marks for Latin characters", () => {
    expect(removeAccentMarks("latté đa ô đê")).toBe("latte da o de");
  });

  test("setUid set an unique string", () => {
    expect(setUid("somename")).toMatchSnapshot();
  });
});
