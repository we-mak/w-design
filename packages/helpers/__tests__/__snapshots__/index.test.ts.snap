// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`object-helpers randomProperty should get the radom value from object 1`] = `"ipsum"`;

exports[`string-helper setUid set an unique string 1`] = `"somename-1577423664661-1"`;
